source/extensions/filters/http/lua/wrappers.h

```diff
@@ -201,6 +203,26 @@ class StreamInfoWrapper : public Filters::Common::Lua::BaseLuaObject<StreamInfoW
   friend class DynamicMetadataMapWrapper;
 };
 
+/**
+ * Lua wrapper for EVP_PKEY.
+ */
+class PublicKeyWrapper : public Filters::Common::Lua::BaseLuaObject<PublicKeyWrapper> {
+public:
+  PublicKeyWrapper(const Envoy::Common::Crypto::PublicKeyPtr* key) : public_key_(key) {}
+  static ExportedFunctions exportedFunctions() { return {{"get", static_luaGet}}; }
+
+  ~PublicKeyWrapper() { delete public_key_; }
+
+private:
+  /**
+   * Get a pointer to public key.
+   * @return pointer to public key.
+   */
+  DECLARE_LUA_FUNCTION(PublicKeyWrapper, luaGet);
+
+  const Envoy::Common::Crypto::PublicKeyPtr* public_key_;
```

Comment:
You should be able to make this just `const Envoy::Common::Crypto::PublicKeyPtr public_key_;`, move the value in constructor, then you don't need `delete` in destructor, (no need of destructor).